test/e2e/functional_test.go

```diff
@@ -84,6 +84,31 @@ spec:
 		ExpectWorkflowDeleted()
 }
 
+func (s *FunctionalSuite) TestWorkflowRetention() {
+	var listOptions metav1.ListOptions
+	for i := 0; i < 15; i++ {
+		s.Given().
+			Workflow(`
+metadata:
+  generateName: workflow-retention-
+spec:
+  ttlStrategy:
+    secondsAfterCompletion: 86400
+  entrypoint: main
+  templates:
+    - name: main
+      container:
+        image: argoproj/argosay:v2
+`).When().SubmitWorkflow()
+	}
+	s.Given().When().
+		Wait(30*time.Second).
```

Comment:
never use `Wait` - the test will always be slow

Reply:
Will use two workflows instead and check for succes instead and then gc.